cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java

```diff
@@ -592,19 +623,33 @@ private MongoDatabase getDatabase(String dbName) {
                 }
                 MongoCredential credential = MongoCredential.createCredential(mongoUsername, authSource,
                         mongoPassword.toCharArray());
+
                 /****
                 // This constructor is deprecated see Mongo Client API documentation
                 // @deprecated Prefer {@link #MongoClient(List, MongoCredential, MongoClientOptions)}
                 client = new MongoClient(servers, Arrays.asList(credential));
                 ****/
                 if ((mongoReplicaSet!= null) && !mongoReplicaSet.isEmpty()) {
                     client = new MongoClient(servers, credential, new MongoClientOptions.Builder().
-                            requiredReplicaSetName(mongoReplicaSet).build());
+                                             requiredReplicaSetName(mongoReplicaSet).
+                                             sslEnabled(sslEnabled).
+                                             sslInvalidHostNameAllowed(sslInvalidHostNameAllowed).
+                                             sslContext(sslContext).
+                                             build());
                 } else {
-                    client = new MongoClient(servers, credential, new MongoClientOptions.Builder().build());
+                    client = new MongoClient(servers, credential, new MongoClientOptions.Builder().
+                                             sslEnabled(sslEnabled).
+                                             sslInvalidHostNameAllowed(sslInvalidHostNameAllowed).
+                                             sslContext(sslContext).
+                                             build());
                 }
             } else {
-                client = new MongoClient(servers);
+                MongoClientOptions options = MongoClientOptions.builder()
```

Comment:
this branch seems extrange, at least "structurally" with respect the previous ones. Could be?

Reply:
Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/2386/commits/d0d92ceaa7202091f9b6f95d7b40296baf7b88dc